### PR TITLE
[FLINK-22364][doc] Translate the page of "Data Sources" to Chinese.

### PR DESCRIPTION
**Pull request by Senhongl**
## What is the purpose of the change

It is aimed to translate the page of "Data Sources" to Chinese, which is still in English currently.

## Brief change log

Translate "docs/content.zh/docs/dev/datastream/sources.md" from English to Chinese.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? no

### CODE REVIEW

**Comment by Senhongl** [comment]
@wuchong @PatrickRen @becketqin 
Could you please help me review it and see if anything needs to be modified?

File: docs/content.zh/docs/dev/datastream/sources.md [hunk 1]
```diff
@@ -154,28 +161,32 @@ class MySplitEnumerator implements SplitEnumerator<MySplit> {
...
-**However, the `SourceReader` implementation should avoid emitting multiple records in a single `pollNext(ReaderOutput)` invocation unless necessary.** This is because the task thread that is polling from the `SourceReader` works in an event-loop and cannot block.
+在创建`源阅读器`时，相关的 `源阅读器上下文` 会提供给 `Source`，而 `Source` 则会将上下文传递给`源阅读器`实例。`源阅读器`可以通过`源阅读器上下文`将 `SourceEvent` 传递给相应的`分片枚举器`。`Source` 的一个典型设计模式是让`源阅读器`报告它们的本地信息给`分片枚举器`，后者通过全局实图来做决定。
 
-All the state of a `SourceReader` should be maintained inside the `SourceSplit`s which are returned at the `snapshotState()` invocation. Doing this allows the `SourceSplit`s to be reassigned to other `SourceReaders` when needed.
+`源阅读器`API 是一个底层(low-level) API，允许用户手动处理splits，并拥有自己的线程模型来获取和移交记录。为了促进`源阅读器`的实现，Flink 提供了 [SourceReaderBase](https://github.com/apache/flink/blob/master/flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/source/reader/SourceReaderBase.java)类，可以显著减少编写 `SourceReader` 所需要的工作量。
```
Comment (PatrickRen) [inline docs/content.zh/docs/dev/datastream/sources.md:182]: 代码中的类名我倾向于不翻译：
`SourceReader` API 是一个底层（low-level）API

Comment (PatrickRen) [inline docs/content.zh/docs/dev/datastream/sources.md:182]: 为了 ~~促进~~ 帮助 `源阅读器`的实现

File: docs/content.zh/docs/dev/datastream/sources.md [hunk 2]
```diff
@@ -208,52 +219,59 @@ val stream = env.fromSource(
...
 
-## The Split Reader API
+<a name="The Split Reader API"></a>
 
-The core SourceReader API is fully asynchronous and requires implementations to manage asynchronous split reading manually.
-However, in practice, most sources perform blocking operations, like blocking *poll()* calls on clients (for example the `KafkaConsumer`), or blocking I/O operations on distributed file systems (HDFS, S3, ...). To make this compatible with the asynchronous Source API, these blocking (synchronous) operations need to happen in separate threads, which hand over the data to the asynchronous part of the reader.
+## 分片阅读器 API
 
-The [SplitReader](https://github.com/apache/flink/blob/master/flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/source/reader/splitreader/SplitReader.java) is the high-level API for simple synchronous reading/polling-based source implementations, like file reading, Kafka, etc.
+核心源阅读器 API是完全异步的，并且要求实现手动管理异步读取分片。
```
Comment (PatrickRen) [inline docs/content.zh/docs/dev/datastream/sources.md:226]: ~~核心~~ 源阅读器 API 是完全异步的

File: docs/content.zh/docs/dev/datastream/sources.md [hunk 3]
```diff
@@ -208,52 +219,59 @@ val stream = env.fromSource(
...
+但是，实际上，大多数sources都会阻塞操作，例如阻塞客户端（如 `KafkaConsumer`）的 *poll()* 调用，或者阻塞分布式文件系统（HDFS, S3等）的I/O操作。为了使其与异步Source API兼容，这些阻塞（同步）操作需要在单独线程中进行，而这些线程会将数据移交给阅读器的异步部分。
```
Comment (PatrickRen) [inline docs/content.zh/docs/dev/datastream/sources.md:227]: 例如阻塞客户端（如 `KafkaConsumer`）的 *poll()* 调用，或者阻塞分布式文件系统（HDFS, S3等）的I/O操作
-> 例如客户端（如 `KafkaConsumer`）的 *poll()* 阻塞调用，或者分布式文件系统（HDFS, S3等）的阻塞 I/O 操作

File: docs/content.zh/docs/dev/datastream/sources.md [hunk 4]
```diff
@@ -333,19 +351,21 @@ public class FixedFetcherSizeSourceReader<E, T, SplitT extends SourceSplit, Spli
...
+## 事件时间和水印
+
+*事件时间* 分配和 *水印生成* 是作为数据源的一部分发生的。离开源阅读器的事件流将具有事件时间戳，并且（在流执行期间）包含水印。有关事件时间和水印的介绍，请参见[及时流处理]({{< ref "docs/concepts/time" >}})。
 
 {{< hint warning >}}
-Applications based on the legacy [SourceFunction](https://github.com/apache/flink/blob/master/flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/source/SourceFunction.java) typically generate timestamps and watermarks in a separate later step via `stream.assignTimestampsAndWatermarks(WatermarkStrategy)`. This function should not be used with the new sources, because timestamps will be already assigned, and it will override the previous split-aware watermarks.
+基于遗留的 [SourceFunction](https://github.com/apache/flink/blob/master/flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/source/SourceFunction.java) 的应用通常在之后的单独的一步中通过 `stream.assignTimestampsAndWatermarks(WatermarkStrategy)` 生成时间戳和水印。这个函数不应该与新的sources一起使用，因为此时时间戳应该已经被分配了，而且该函数会覆盖掉之前的可识别分片水印。
```
Comment (PatrickRen) [inline docs/content.zh/docs/dev/datastream/sources.md:363]: 基于 ~~遗留~~ 旧 的

File: docs/content.zh/docs/dev/datastream/sources.md [hunk 5]
```diff
@@ -354,32 +374,36 @@ environment.fromSource(
...
-Watermark Generators are only active during streaming execution. Batch execution deactivates Watermark Generators; all related operations described below become effectively no-ops.
+水印生成器仅在流执行期间会被激活。批处理执行则会停用水印生成器，则下文所述的所有相关操作实际上都变为无操作。
 
-The data source API supports running watermark generators individually *per split*. That allows Flink to observe the event time progress per split individually, which is important to handle *event time skew* properly and prevent *idle partitions* from holding back the event time progress of the entire application.
+数据源API支持*每个分片*单独运行水印生成器。这使得Flink可以分别观察每个分片的事件时间进度，这对于正确处理*事件时间偏斜*和防止*空闲分区*阻碍整个应用程序的事件时间进度是很重要的。
 
 {{< img width="80%" src="fig/per_split_watermarks.svg" alt="Watermark Generation in a Source with two Splits." >}}
 
-When implementing a source connector using the *Split Reader API*, this is automatically handled. All implementations based on the Split Reader API have split-aware watermarks out-of-the-box.
+使用*分片提取器 API*实现源连接器时，将自动进行处理。所有基于分片提取器 API的实现都具有开箱即用（out-of-the-box）的可识别分片的水印。
```
Comment (PatrickRen) [inline docs/content.zh/docs/dev/datastream/sources.md:407]: 使用*分片提取器 API*实现源连接器时
-> 使用*分片阅读器（SplitReader）API* 实现源连接器时

与上文的翻译保持一致

File: docs/content.zh/docs/dev/datastream/sources.md [hunk 6]
```diff
@@ -354,32 +374,36 @@ environment.fromSource(
...
 
-For an implementation of the lower level `SourceReader` API to use split-aware watermark generation, the implementation must output events from different splits to different outputs: the *Split-local SourceOutputs*. Split-local outputs can be created and released on the main [ReaderOutput](https://github.com/apache/flink/blob/master/flink-core/src/main/java/org/apache/flink/api/connector/source/ReaderOutput.java) via the `createOutputForSplit(splitId)` and `releaseOutputForSplit(splitId)` methods. Please refer to the JavaDocs of the class and methods for details.
+为了实现较低级别的`源阅读器`API可以使用可识别分区的水印生成，必须将事件从不同的分片输出到不同的输出中：*局部分片 SourceOutputs*。通过 `createOutputForSplit(splitId)` 和 `releaseOutputForSplit(splitId)` 方法，可以在总 [ReaderOutput](https://github.com/apache/flink/blob/master/flink-core/src/main/java/org/apache/flink/api/connector/source/ReaderOutput.java) 上创建并发布局部分片输出。有关详细信息，请参阅该类和方法的JavaDocs。
```
Comment (PatrickRen) [inline docs/content.zh/docs/dev/datastream/sources.md:409]: 为了实现 ~~较低级别~~ 更底层 的`源阅读器`API可以使用可识别分区的水印生成

**Review by PatrickRen — COMMENTED** [review]
Thanks for the contribution Senhong! I left some comments for the translation.

**Comment by Senhongl** [comment]
@PatrickRen Hi, I have updated the document based on your advice, please take a look.

**Review by wuchong — APPROVED** [review]
LGTM.

**Pull request merged**